src/snowflake/snowpark/dataframe.py

```diff
@@ -343,6 +349,143 @@ def _disambiguate(
     return lhs_remapped, rhs_remapped
 
 
```

Comment:
Should we add docstring to introduce this improvement? I think it will be very helpful if users can find it in documentation.

Reply:
Yeah. I'll add that. thanks for calling it out